py/mkenv.mk

```diff
@@ -59,18 +59,28 @@ LD += -m32
 endif
 
 MAKE_FROZEN = $(TOP)/tools/make-frozen.py
-# allow mpy-cross (for WSL) and mpy-cross.exe (for cygwin) to coexist
+# allow mpy-cross (for WSL, Cygwin or Linux) and mpy-cross.exe (for Windows / MSYS2 / mingw) to coexist
```

Comment:
This is not entirely correct: the .exe or not does not depend on the host 'OS" (or emulation layer or whatever it's called), but ion the compiler used. So if you'd use gcc on Cygwin (or MSYS2 possibly) you don't want to end up with .exe, but on those same hosts you build with mingw-w64 you do want to end up with .exe.

Reply:
mpy-cross is used while running "make". So it should be compiled to run in the same environment that the makefile is running in.
> So if you'd use gcc on Cygwin (or MSYS2 possibly) you don't want to end up with .exe

You won't. All POSIX shells report their own OS name, eg msys. You only get Windows-NT from a raw Windows cmd.exe or a Windows program such as Eclipse.  Will change line 62:

`# allow mpy-cross (for WSL, MSYS, Cygwin or Linux shell) and mpy-cross.exe (for Windows cmd.exe) to coexist`

Reply:
I build under Cygwin using ```make CROSS_COMPILE=i686-w64-mingw32-``` and get a .exe.